server/cluster/cluster.go

```diff
@@ -1071,6 +1072,12 @@ func (c *RaftCluster) RemoveStore(storeID uint64, physicallyDestroyed bool) erro
 		return errs.ErrStoreDestroyed.FastGenByArgs(storeID)
 	}
 
+	if (store.IsPreparing() || store.IsServing()) && !physicallyDestroyed {
+		if err := c.checkReplicaBeforeOfflineStore(storeID); err != nil {
```

Comment:
When the placement rule is enabled and we cannot easily tell if it should be allowed or not, I think we should always return no error and allow the operation.

Reply:
However, we think removing the store directly when the placement rule is enabled is not safe, and if the client does want to offline the store even when the number of up stores is not enough and the placement rule is enabled, he can still remove the store with `physicallyDestroyed` true